src/display/editor/tools.js

```diff
@@ -1469,6 +1475,123 @@ class AnnotationEditorUIManager {
     }
   }
 
+  /**
+   * Set up the drag session for moving the selected editors.
+   */
+  setUpDragSession() {
+    if (!this.hasSelection) {
+      return;
+    }
+    // Avoid to have spurious text selection in the text layer when dragging.
+    this.disableUserSelect(true);
+    this.#draggingEditors = new Map();
+    for (const editor of this.#selectedEditors) {
+      this.#draggingEditors.set(editor, {
+        savedX: editor.x,
+        savedY: editor.y,
+        savedPageIndex: editor.parent.pageIndex,
+        newX: 0,
+        newY: 0,
+        newPageIndex: -1,
+      });
+    }
+  }
+
+  /**
+   * Ends the drag session.
+   * @returns {boolean} true if at least one editor has been moved.
+   */
+  endDragSession() {
+    if (!this.#draggingEditors) {
+      return false;
+    }
+    this.disableUserSelect(false);
+    const map = this.#draggingEditors;
+    this.#draggingEditors = null;
+    let mustBeAddedInUndoStack = false;
+
+    for (const [{ x, y, parent }, value] of map) {
+      value.newX = x;
+      value.newY = y;
+      value.newParentPageIndex = parent.pageIndex;
```

Comment:
That property doesn't seem to exist, did you actually mean the following?
```suggestion
      value.newPageIndex = parent.pageIndex;
```